### PR TITLE
Handle physical action-triggered outputs in federated execution TS (previously handled by TAN - Time Advance Notice)

### PR DESCRIPTION
**Pull request by hokeun**
Related issue: https://github.com/lf-lang/reactor-ts/issues/94

Related PR: https://github.com/lf-lang/lingua-franca/pull/1275

This will unblock the progress of the Chat Application example in TypeScript (where the chat user is blocked indefinitely due to the existence of physical action in the upstream federate.): https://github.com/lf-lang/examples-lingua-franca/pull/10

### CODE REVIEW

File: src/core/time.ts [hunk 1]
```diff
@@ -134,6 +134,19 @@ export class TimeValue {
         return TimeValue.secsAndNs(s, ns);
     }
 
+    multiply(factor: number): TimeValue {
```
Comment (hokeun) [inline src/core/time.ts:137]: Add a unit test for this.

Comment (hokeun) [reply src/core/time.ts:137]: Added it here: https://github.com/lf-lang/reactor-ts/pull/106/commits/1aebda4ccb93d4add5f5936492ed44ea3faacd6d

**Review by lhstrh — APPROVED** [review]

**Review by Soroosh129 — APPROVED** [review]
LGTM!

**Comment by hokeun** [comment]
Thank you @lhstrh and @Soroosh129 !

File: src/core/federation.ts [hunk 2]
```diff
@@ -835,7 +843,16 @@ export class FederatedApp extends App {
             let greatestTAG = this._getGreatestTimeAdvanceGrant();
             let nextTag = event.tag;
             if (greatestTAG === null || greatestTAG.isSmallerThan(nextTag)) {
-                this.sendRTINextEventTime(nextTag);
+                if (this.minDelayFromPhysicalActionToFederateOutput !== null &&
+                    this.downstreamFedIDs.length > 0) {
+                        let physicalTime = getCurrentPhysicalTime()
+                        if (physicalTime.add(this.minDelayFromPhysicalActionToFederateOutput).isEarlierThan(event.tag.time)) {
+                            Log.debug(this, () => "Adding dummy event for tag: " + event.tag);
+                            this._addDummyEvent(event.tag);
```
Comment (Soroosh129) [inline src/core/federation.ts:851]: One question: shouldn't this be `this._addDummyEvent(physicalTime.add(this.minDelayFromPhysicalActionToFederateOutput))`?

Comment (hokeun) [reply src/core/federation.ts:851]: Thank you for catching this! This PR https://github.com/lf-lang/reactor-ts/pull/109 should address this issue.

**Pull request merged**